test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,288 @@
+        self.assertEqual(mask, torch.tensor([[[ True, False]],
+                                             [[ True, True]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2]))
+        self.assertEqual(mask.dim(), 3)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+        self.assertEqual(tensor.dim(), original_nt2.dim())
+
+        # mask_dim = 0
+        tensor, mask = original_nt2.to_tensor_mask(mask_dim=0)
+        self.assertEqual(mask, torch.tensor([[[ True, False]],
+                                             [[ True, True]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2]))
+        self.assertEqual(mask.dim(), 3)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+        self.assertEqual(tensor.dim(), original_nt2.dim())
+
+        # mask_dim = 1
+        tensor, mask = original_nt2.to_tensor_mask(mask_dim=1)
+        self.assertEqual(mask, torch.tensor([[[ True, False]],
+                                             [[ True, True]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2]))
+        self.assertEqual(mask.dim(), 3)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+
+        # mask_dim = 2
+        tensor, mask = original_nt2.to_tensor_mask(mask_dim=2)
+        self.assertEqual(mask, torch.tensor([[[ True, False]],
+                                             [[ True,  True]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2]))
+        self.assertEqual(mask.dim(), 3)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+        self.assertEqual(tensor.dim(), original_nt2.dim())
+
+        # mask_dim = 3
+        tensor, mask = original_nt2.to_tensor_mask(mask_dim=3)
+        self.assertEqual(mask, torch.tensor([[[ True, False]],
+                                             [[ True, True]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2]))
+        self.assertEqual(mask.dim(), 3)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+        self.assertEqual(tensor.dim(), original_nt2.dim())
+
+        # mask_dim = 4
+        tensor, mask = original_nt2.to_tensor_mask(mask_dim=4)
+        self.assertEqual(mask, torch.tensor([[[[ True, True],
+                                               [False, False]]],
+                                             [[[ True, True],
+                                               [ True, True]]]]))
+        self.assertEqual(mask.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(mask.dim(), 4)
+        self.assertEqual(tensor.size(), torch.Size([2, 1, 2, 2]))
+        self.assertEqual(tensor.dim(), 4)
+        self.assertEqual(tensor.dim(), original_nt2.dim())
+
+    def test_to_tensor_mask_nt_dim_5(self):
```

Comment:
with all this repetition it seems worth writing a local generator function or even something more abstract

Reply:
Ok, will refactor to be nicer. 
I personally prefer writing test code like this cause its immediately obvious what is failing and why. But i can see why this is not ideal.

Reply:
I still don't have a 100% trust for `assertEqual` and just wanted to make sure that all the expectations are correct and my understanding of the expected result is right. Will refactor but it would be nice if you could take a quick look at few random cases and verify the expectations.